vars/commonPipelineEnvironment.groovy

```diff
@@ -192,36 +192,31 @@ class commonPipelineEnvironment implements Serializable {
     ]
 
     void writeToDisk(script) {
-
         files.each({f  ->
-            if (this[f.property] && !script.fileExists(f.filename)) {
-                script.writeFile file: f.filename, text: this[f.property]
-            }
+            writeValueToFile(script, f.filename, this[f.property])
         })
 
         containerProperties.each({key, value ->
-            def fileName = ".pipeline/commonPipelineEnvironment/container/${key}"
-            if (value && !script.fileExists(fileName)) {
-                if(value in CharSequence) {
-                    script.writeFile file: fileName, text: value
-                } else {
-                    script.writeFile file: fileName, text: groovy.json.JsonOutput.toJson(value)
-                }
-            }
+            writeValueToFile(script, ".pipeline/commonPipelineEnvironment/container/${key}", value)
         })
 
         valueMap.each({key, value ->
-            def fileName = ".pipeline/commonPipelineEnvironment/custom/${key}"
-            if (value && !script.fileExists(fileName)) {
-                if(value in CharSequence) {
-                    script.writeFile file: fileName, text: value
-                } else {
-                    script.writeFile file: fileName, text: groovy.json.JsonOutput.toJson(value)
-                }
-            }
+            writeValueToFile(script, ".pipeline/commonPipelineEnvironment/custom/${key}", value)
         })
     }
 
+    void writeValueToFile(script, String filename, value){
+        if (value == null)
+            return
+        if (!value in CharSequence) {
+            filename += '.json'
```

Comment:
I understand that a file containing some format (... here: json) should be named accordingly (... here: with the appropriate `.json` appendix). That is new and seems to be the real change.

But I would expect that the file appendix does not really matter. Even if the file does not come with a `.json` appendix it should be possible to parse the content of the file in the same way like when the file comes with that appendix.

--> why it is important to add that appendix here. What is really fixed hereby?

Reply:
The `readFromDisk` function needs the suffix to handle the content appropriately. Any idea how to handle this in a better way?

https://github.com/SAP/jenkins-library/blob/647704e516e850abb75a040edac2155c90638244/vars/commonPipelineEnvironment.groovy#L232

Reply:
I think this was the missing piece. When reading back we make a guess based on the appendix (... which is quite common).

>Any idea how to handle this in a better way?

No need to improve that IMO. With xml it would be possible to check based on the xml declaration, but there is nothing similar for json, yml ...

Maybe we it would be a good idea to extract the `.json` to a variable and use that when writing/reading. With that the story here would be easier to understand.